source/_docs/automation/modes.markdown

```diff
@@ -51,3 +51,42 @@ automation:
     action:
       - ...
 ```
+
+When choosing the correct mode take into consideration that top level triggers only trigger once for any single instance of an automation. If your automation has until loops that rely on Trigger IDs consider using the restart mode. In this example if we set the mode to single the release trigger will not stop the light decreasing as the automation will be in the repeat loop. 
+
+## Example of restart
+
+```yaml
```

Comment:
Please adjust to being more compact / more to the point / without optional values / empty defaults.

Also, make sure it complies with our YAML Style Guide: <https://developers.home-assistant.io/docs/documenting/yaml-style-guide>